pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -112,7 +116,8 @@ func identityMatches(set *apps.StatefulSet, pod *v1.Pod) bool {
 	return ordinal >= 0 &&
 		set.Name == parent &&
 		pod.Name == getPodName(set, ordinal) &&
-		pod.Namespace == set.Namespace
+		pod.Namespace == set.Namespace &&
+		pod.Labels[StatefulSetPodNameLabel] == pod.Name
```

Comment:
Do we lose anything by making the label just the ordinal, instead of the whole Pod name?

I think it's more flexible. For example, you could select the ordinal=0 Pod across multiple StatefulSets.

Reply:
yes, the pod name is namespace unique. The ordinal is not.

Reply:
Selecting the same Pod across multiple sts's defeats the purpose of the label.

Reply:
I see. The ordinal label would be insufficient, even if combined with the other selector labels, because controllers can have overlapping selectors. Agreed that selecting across multiple StatefulSets doesn't meet any need we've ever heard of.